tests/unit/moduleapi/crash.tcl

```diff
@@ -0,0 +1,50 @@
+# This file is used to test certain crash edge cases to make sure they produce
+# correct stack traces for debugging.
+set testmodule [file normalize tests/modules/crash.so]
+
+# Valgrind will complain that the process terminated by a signal, skip it.
+if {!$::valgrind} {
+    start_server {tags {"modules"}} {
+        r module load $testmodule assert
+        test {Test module crash when info crashes with an assertion } {
+            catch {r 0 info infocrash}
+            set res [wait_for_log_messages 0 {"*=== REDIS BUG REPORT START: Cut & paste starting from here ===*"} 0 10 1000]
+            set loglines [lindex $res 1]
+
+            set res [wait_for_log_messages 0 {"*ASSERTION FAILED*"} $loglines 10 1000]
+            set loglines [lindex $res 1]
+
+            set res [wait_for_log_messages 0 {"*RECURSIVE ASSERTION FAILED*"} $loglines 10 1000]
+            set loglines [lindex $res 1]
+
+            wait_for_log_messages 0 {"*=== REDIS BUG REPORT END. Make sure to include from START to END. ===*"} $loglines 10 1000
+            assert_equal 1 [count_log_message 0 "=== REDIS BUG REPORT END. Make sure to include from START to END. ==="]
+            assert_equal 2 [count_log_message 0 "ASSERTION FAILED"]
+            assert_equal 1 [count_log_message 0 "RECURSIVE ASSERTION FAILED"]
+            assert_equal 1 [count_log_message 0 "=== REDIS BUG REPORT START: Cut & paste starting from here ==="]
+        }
+    }
+
+    start_server {tags {"modules"}} {
+        r module load $testmodule segfault
+        test {Test module crash when info crashes with a segfault} {
```

Comment:
maybe we can unify these two tests with a loop and some conditions on the non-common parts?
OTOH, it doesn't look like it'll reduce much, and help in readability...

Reply:
Yeah, I kept it as is mostly since I think it makes it more readable.